pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -456,3 +474,116 @@ func (cloud *CloudProvider) GetInstanceOrError(ctx context.Context, instanceZone
 	glog.V(4).Infof("Got instance %v from zone %v", instanceName, instanceZone)
 	return instance, nil
 }
+
+
+func (cloud *CloudProvider) GetSnapshot(ctx context.Context, snapshotName string) (*compute.Snapshot, error) {
+	svc := cloud.service
+	project := cloud.project
+	glog.V(4).Infof("Getting snapshot %v", snapshotName)
+	snapshot, err := svc.Snapshots.Get(project, snapshotName).Context(ctx).Do()
+	if err != nil {
+		return nil, err
+	}
+	glog.V(4).Infof("Got snapshot %v", snapshotName)
+	return snapshot, nil
+}
+
+func (cloud *CloudProvider) DeleteSnapshot(ctx context.Context, snapshotName string) error {
+	op, err := cloud.service.Snapshots.Delete(cloud.project, snapshotName).Context(ctx).Do()
+	if err != nil {
+		if IsGCEError(err, "notFound") {
+			// Already deleted
+			return nil
+		}
+		return err
+	}
+	err = cloud.waitForGlobalOp(ctx, op)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cloud *CloudProvider) CreateSnapshot(ctx context.Context, volKey *meta.Key, snapshotName string) (*compute.Snapshot, error) {
+	switch volKey.Type() {
+	case meta.Zonal:
+		snapshot, err := cloud.createSnapshotOfZonalDisk(ctx, volKey, snapshotName)
```

Comment:
nit: `createZonalDiskSnapshot`? and you can just return on the same line here: `return cloud....`

Reply:
done